File: consentor-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__ConsentorDescriptor.java
package ${package}.${pluginName};

import se.curity.identityserver.sdk.oauth.consent.Consentor;
import se.curity.identityserver.sdk.plugin.descriptor.ConsentorPluginDescriptor;

public final class ${pluginName}ConsentorDescriptor implements ConsentorPluginDescriptor<${pluginName}ConsentorConfig> {

    @Override
    public Class<? extends Consentor> getConsentorType()
    {
        return ${pluginName}Consentor.class;
    }

    @Override
    public String getPluginImplementationType()
    {
        return "${pluginName.toLowerCase()}";
    }

    @Override
    public Class<? extends ${pluginName}ConsentorConfig> getConfigurationType()
    {
        return ${pluginName}ConsentorConfig.class;
    }
}

File: kotlin-consentor-archetype/src/main/resources/archetype-resources/src/main/kotlin/__pluginName__/__pluginName__ConsentorDescriptor.kt
package ${package}.${pluginName}

import se.curity.identityserver.sdk.plugin.descriptor.ConsentorPluginDescriptor

class ${pluginName}ConsentorDescriptor: ConsentorPluginDescriptor<${pluginName}ConsentorConfig>
{
    override fun getConsentorType() = ${pluginName}Consentor::class.java

    override fun getPluginImplementationType() = "${pluginName.toLowerCase()}"

    override fun getConfigurationType() = ${pluginName}ConsentorConfig::class.java
}
